src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -202,18 +202,18 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
 
             // Have to store the snapshot to reuse it in some projections related scenarios
             // where data and session in further calls are able to provide other snapshots.
-            session.Properties.AddProperty(TriggerSnapshot, triggerLocation.Snapshot);
+            session.Properties[TriggerSnapshot] = triggerLocation.Snapshot;
```

Comment:
i feel like this PR isn't really explaining why this fix is appropriate.  How are we getting in the situation where we're adding this property multiple times?  This feels more like a workaround versus this being a fix of a root problem.

Reply:
>  We could either check for the case and throw a NFW but Editor asked us to avoid crashes saying that they will manage the rest on their side. @AmadeusW to confirm.

Hrmm... 
1. i would just prefer the editor take care of this.
2. my preference about how one component handle  bugs in another is generally to bail out, not try to continue in some resilient fashion.

That said, i don't feel very strongly about this.  However, i *would* appreciate if the code was documented with an explanation of why we're doin this.  Basically, that we shouldn't have to, but this works around some bug (with a vstfdevdiv link to the editor bug tracking this issue).